core/http.md

```diff
@@ -0,0 +1,3293 @@
+      ... Resource entity excluded for brevity ...
+    } *
+  } *
+}
+```
+
+**Examples:**
+
+Create a Message and a Schema Resource under Group `g1`:
+
+```yaml
+POST /groups/g1
+Content-Type: application/json; charset=utf-8
+
+{
+  "messages": {
+    "messageid": "msg1",
+    ... remainder of msg1 definition excluded for brevity ...
+  },
+  "schemas": {
+    "schemaid": "schema1",
+    ... remainder of schema1 definition excluded for brevity ...
+  }
+}
+```
+
+```yaml
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+
+{
+  "messages": {
+    "messageid": "msg1",
+    ... remainder of msg1 definition excluded for brevity ...
+  },
+  "schemas": {
+    "schemaid": "schema1",
+    ... remainder of schema1 definition excluded for brevity ...
+  }
+}
+```
+
+#### `DELETE /<GROUPS>/<GID>`
+
+A server MAY support clients deleting a
+[Group](./spec.md#group-entity) via an HTTP `DELETE` directed to the Group
+entity.
+
+The processing of this API is defined in the
+[Deleting Entities](./spec.md#deleting-entities) section of the
+[core specification](./spec.md).
+
+The request MUST be of the form:
+
+```yaml
+DELETE /<GROUPS>/<GID>
+```
+
+A successful response MUST be of the form:
+
+```yaml
+HTTP/1.1 200 OK
+or
+HTTP/1.1 204 No Content
+```
+
+**Examples:**
+
+Delete a Group:
+
+```yaml
+DELETE /endpoints/ep1
+```
+
+```
+HTTP/1.1 204 No Content
+```
+
+Delete a Group, verifying its `epoch` value:
+
+```yaml
+DELETE /endpoints/ep1?epoch=5
+```
+
+```
+HTTP/1.1 204 No Content
+```
+
+### Resource Entity
+
+#### Resource Metadata vs Resource Document
+
+In the core specification's
+[Resource Metadata vs Resource Document](./spec.md#resource-metadata-vs-resource-document)
+section, it explains how Resources might have a domain-specific document
+associated with them via the
+[`hasdocument` model aspect](./model.md#groupsstringresourcesstringhasdocument`).
+For HTTP, clients indicate which view of the Resource they want to interact
+with via the use of a `$details` suffix on the URL to the Resource. Presence
+of the `$details` suffix MUST be interpreted as a request to interact with
+the xRegistry metadata view of the Resource, while its absence MUST be
+interpreted as a request to interact with the Resource's domain-specific
+document.
+
+For example:
+
+```yaml
+http://registry.example.com/schemagroups/mygroup/schemas/myschema$details
+```
+
+references the xRegistry metadata of the `myschema` Resource, while:
+
+```yaml
+http://registry.example.com/schemagroups/mygroup/schemas/myschema
+```
+
+references the (domain-specific) schema document associated with the
+`myschema` Resource.
+
+If a Resource does not support domain-specific documents (i.e.
+[`hasdocument`](./model.md#groupsstringresourcesstringhasdocument)
+is set to `false`), then use of the `$details` suffix in a request URL MUST
+be treated the same as if it were absent and any URLs in server response
+messages referencing that Resource (e.g. `self`) MUST NOT include it.
```

Comment:
This feels odd to me. It means that if a resource changes from not having a domain-specific document to having one, a response can change from metadata to document... rather than changing from error to document.

This may all be ancient history and decided by now, but it feels to me like if I ask for something (the document) and that's not supported, that should give an error, rather than a different result.

To put it another way - suppose we'd made the metadata the default, and had `$document` as a suffix when requesting a document, would we want an explicit `$document` suffix request to return the metadata if there was no document?

Reply:
> Fundamentally, it feels like this approach reduces predictability. If I see a URL which doesn't have $details, I don't know whether that will return the document or the metadata, unless I know whether the resource supports documents or not.

Yup that's true, and to some extent it's done on purpose. As i mentioned, we wanted the naked URL to always be "the data that's most important to THAT client". And the answer to that question changes based on whether the Resource type has a doc or not. So, here, "usability for client of THIS Resource type" takes precedence over "consistency between doc-Resources and non-doc-Resources".

Let's discuss on the call